Sources/TuistGenerator/Mappers/SynthesizedResourceInterfaceProjectMapper.swift

```diff
@@ -49,45 +70,35 @@ public final class SynthesizedResourceInterfaceProjectMapper: ProjectMapping { /
 
         var sideEffects: [SideEffectDescriptor] = []
 
-        let assetsSideEffects: [SideEffectDescriptor]
-        (target, assetsSideEffects) = try renderAndMapTarget(
-            .assets,
-            target: target,
-            project: project
-        )
-        sideEffects += assetsSideEffects
-
-        let stringsSideEffects: [SideEffectDescriptor]
-        (target, stringsSideEffects) = try renderAndMapTarget(
-            .strings,
-            target: target,
-            project: project
-        )
-        sideEffects += stringsSideEffects
-
-        let plistsSideEffects: [SideEffectDescriptor]
-        (target, plistsSideEffects) = try renderAndMapTarget(
-            .plists,
-            target: target,
-            project: project
-        )
-        sideEffects += plistsSideEffects
-
-        let fontsSideEffects: [SideEffectDescriptor]
-        (target, fontsSideEffects) = try renderAndMapTarget(
-            .fonts,
-            target: target,
-            project: project
-        )
-        sideEffects += fontsSideEffects
+        try project.resourceSynthesizers
+            .map { resourceSynthesizer throws -> (ResourceSynthesizer, String) in
+                switch resourceSynthesizer.template {
+                case let .file(path):
+                    let templateString = try FileHandler.shared.readTextFile(path)
+                    return (resourceSynthesizer, templateString)
+                case .defaultTemplate:
+                    return (resourceSynthesizer, try templateString(for: resourceSynthesizer.parser))
+                }
+            }
+            .forEach { parser, templateString in
+                let interfaceTypeEffects: [SideEffectDescriptor]
+                (target, interfaceTypeEffects) = try renderAndMapTarget(
+                    parser,
+                    templateString: templateString,
+                    target: target,
+                    project: project
+                )
+                sideEffects += interfaceTypeEffects
+            }
```

Comment:
nit: can probably convert this to `reduce` so you can make `sideEffects` constant.